modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/BinarySpatialOpTypeBinding.java

```diff
@@ -55,6 +62,18 @@ public Class getType() {
     
     @Override
     public Object getProperty(Object object, QName name) throws Exception {
-        return FESParseEncodeUtil.getProperty((BinarySpatialOperator) object, name);
+        Expression e = FESParseEncodeUtil.getProperty((BinarySpatialOperator) object, name);
+        if (e instanceof Literal && ((Literal) e).getValue() instanceof Geometry) {
+            return new EncoderDelegate() {
+                @Override
+                public void encode(ContentHandler output) throws Exception {                    
+                    Encoder encoder = new Encoder(new org.geotools.gml3.v3_2.GMLConfiguration());
```

Comment:
FES 2.0 has no explicit dependency on GML 3.2.1, or any GML version for that matter. Why should GML 3.2.1 be used here? FES 2.0 can be used to encode filters with any GML version.